source/simulator/introspection.cc

```diff
@@ -312,6 +312,30 @@ namespace aspect
 
 
 
+  template <int dim>
+  bool
+  Introspection<dim>::composition_type_exists (const typename Parameters<dim>::CompositionalFieldDescription::Type &type) const
+  {
+    for (unsigned int c=0; c<composition_descriptions.size(); ++c)
+      if (composition_descriptions[c].type == type)
+        return true;
+    return false;
+  }
+
+
+
+  template <int dim>
+  unsigned int
+  Introspection<dim>::find_composition_type (const typename Parameters<dim>::CompositionalFieldDescription::Type &type) const
+  {
+    for (unsigned int c=0; c<composition_descriptions.size(); ++c)
+      if (composition_descriptions[c].type == type)
+        return c;
+    return composition_descriptions.size();
```

Comment:
do we want to assert here instead?

Reply:
I implemented it like this because that's equivalent to the `std::find` function one would normally use for a vector, and it has the advantage that it doesn't crash if the field does not exist. 

But I am not feeling strongly about this; if you think it's better with an assert then I can change it.